### PR TITLE
chore: refactor hooks : moved few hooks in right folder

### PR DESCRIPTION
**Pull request by NikitaCG**
- useSelect
 - useTimeout
 - useViewportSize
 - useListNavigation
 - useIntersection

### CODE REVIEW

File: src/hooks/useViewportSize/useViewportSize.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import React from 'react';
 
-export interface ViewportSize {
+export interface ViewportSizeResult {
```
Comment (amje) [inline src/hooks/useViewportSize/useViewportSize.tsx:3]: We should keep this type as is, and create `UseViewportSizeResult` type as an alias to this type

**Review by amje — CHANGES_REQUESTED** [review]

File: src/hooks/useViewportSize/useViewportSize.tsx (hunk 1 above)
Comment (NikitaCG) [reply src/hooks/useViewportSize/useViewportSize.tsx:3]: Up to this point, we haven't exported this hook yet, so no one can use it. That's why I renamed without an alias. Are you thinking of making an alias after all?

Comment (amje) [reply src/hooks/useViewportSize/useViewportSize.tsx:3]: Then `Use` should be at the beginning

Comment (NikitaCG) [reply src/hooks/useViewportSize/useViewportSize.tsx]: done

File: src/hooks/useViewportSize/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+```tsx
+import {useViewportSize} from '@gravity-ui/uikit';
+```
+
+The `useViewportSize` hook calculates the size of the viewport at the resize event
+
+## Result
+
+|  Name  |   Description   |   Type   |
+| :----: | :-------------: | :------: |
```
[the remark this answers is not in the document]
Comment (NikitaCG) [reply src/hooks/useViewportSize/README.md]: fixed

**Review by amje — APPROVED** [review]

**Pull request merged**